ClothesApp/Application/Repositories/IBaseRepository.cs

```diff
@@ -0,0 +1,18 @@
+namespace Application.Repositories;
+
+public interface IBaseRepository<T>
```

Comment:
Generic repository idea is good, but what will you do if for some entities there will be no way to delete them or there will be no name to check for (this is about ExistAsync(string name)?